src/mprk.jl

```diff
@@ -998,7 +998,7 @@ struct MPRK43II{T, F, T2} <: OrdinaryDiffEqAdaptiveAlgorithm
     small_constant_function::T2
 end
 
-function MPRK43II(gamma; linsolve = LUFactorization(), small_constant = nothing)
+function MPRK43II(gamma; linsolve = LUFactorization(), small_constant = 1e-50)
```

Comment:
Did you check how it works for Float32` computations? Do we convert the `Float64` value from here appropriately?

Reply:
No, we don't. Setting `small_constant = (type) -> type(1e-50)` would generally work. But `Float32(1e-50)=0.0f0`, which doesn't help to avoid divisions by zero. I guess we should use an if statement and set the value of `small_constant` regarding to the data type.

We don't have any tests for `Float32` computations. Do we need those?

Reply:
I think it would be nice to check some computations with `Float32`.

Reply:
My answer was not specific enough. I'm also not sure what the desired behavior is. 

If we create a problem with `Float32` data like

```julia
julia> using PositiveIntegrators

julia> P_linmod(u, p, t) = [0 u[2]; 5*u[1] 0]
P_linmod (generic function with 1 method)

julia> u0 = [0.9f0, 0.1f0]
2-element Vector{Float32}:
 0.9
 0.1

julia> prob = ConservativePDSProblem(P_linmod, u0, (0.0f0, 2.0f0))
ODEProblem with uType Vector{Float32} and tType Float32. In-place: false
timespan: (0.0f0, 2.0f0)
u0: 2-element Vector{Float32}:
 0.9
 0.1
```
and also use `Float32` parameters as inputs for the algorithms, then the problem can be solved. 
```
julia> solve(prob, MPRK22(1.0f0); dt=0.1, save_everystep=false)
retcode: Success
Interpolation: 1st order linear
t: 2-element Vector{Float64}:
 0.0
 2.0
u: 2-element Vector{Vector{Float32}}:
 [0.9, 0.1]
 [0.16667159, 0.83332926]
```
But although the output of `ConservativePDSProblem` said `tType` is `Float32` we now see that `sol.t` is a `Vector{Float64}`. I guess this is a bug, right?

Now we focus solely on `sol.u`. `MPE` which doesn't have any parameters can be used as well. 
```julia
julia> solve(prob, MPE(); dt=0.1, save_everystep=false)
retcode: Success
Interpolation: 1st order linear
t: 2-element Vector{Float64}:
 0.0
 2.0
u: 2-element Vector{Vector{Float32}}:
 [0.9, 0.1]
 [0.16672719, 0.83327186]
```
But using `Float64` inputs like in `solve(prob, MPRK22(1.0); dt=0.1, save_everystep=false)` will throw an error.  The reason for the error is that currently `small_constant` which is by default `floatmin(uEltypeNoUnits)` must have the same data type as the other constants in the constant's cache. 

Is this the behavior we want?

In `SSPMPRK43` the constants are hard coded in `Float64` and currently 'small_constant=1e-50` for this algorithm.  So the following works as well, because everything in `SSPMPRK43ConstantCache` has the same data type.
```julia
julia> solve(prob, SSPMPRK43(); dt=0.1, save_everystep=false)
retcode: Success
Interpolation: 1st order linear
t: 2-element Vector{Float64}:
 0.0
 2.0
u: 2-element Vector{Vector{Float32}}:
 [0.9, 0.1]
 [0.16667144, 0.83332855]
```
Do we want to convert the constants to `Float32` or are we fine with this implicit conversion?

Reply:
As a user, I would expect to get purely `Float32` computations everywhere when I set up my problem with `Float32` data and solve it using default settings. So yes, we should check why the time is converted to `Float64` and we should take care of converting the `small_constant`s.

Reply:
To be sure, we could check hardware performance counters (e.g., LIKWID.jl) locally. There should only be single precision computations, nothing with double precision, if the problem is specified using `Float32`.

Reply:
I just came back to this. In my comment above I made the crucial mistake to use `dt=0.1` instead of `dt=0.1f0`. If `dt` is a `Float32` , then `sol.t` is a `Vector{Float32}` as expected.

Reply:
> Did you check how it works for Float32`computations? Do we convert the`Float64` value from here appropriately?

This should be fine how. Conversion was added.